### ISSUE TITLE
Mapping of path handlers

### ISSUE THREAD

**Issue by Somebi** [APIDevTools/swagger-express-middleware#148]
There is no automated mapping for path request handlers?
If there is for example regexp logic for concrete path provided in swagger specification file, it should be also manually reflected in the code by doing app[path_name](handler)?
Or you are providing some method, which accepts handlers by their operationId which is provided in the swagger specification file?

**Comment by Somebi** [comment]
Because there are differences between how express handles routes /user/:id and swagger /user/{id}

**Comment by jhenaoz** [comment]
I have the same issue:

/pets/{id} PATCH, DELETE --> id, integer
/pets/counts GET

And when I call pets/counts, swagger middleware think is the first one, and express think is the second one, so with swagger, is a bad request and method not allowed, it should be better to try first exact match instead of path variables.

**Comment by JamesMessinger** [comment]
Please see [the docs for the Parse Request middleware](https://apidevtools.org/swagger-express-middleware/docs/middleware/parseRequest.html), which explains how the incoming request is parsed and mapped to the corresponding Swagger components.

Specifically, see [Phase 3 - Path parsing](https://apidevtools.org/swagger-express-middleware/docs/middleware/parseRequest.html#phase-3---path-parsing) which explains how to map Express path parameters (e.g. `/user/:id`) to Swagger path parameters (e.g. `/user/{id}`).  

**tldr:** Use `req.pathParams` instead of `req.params`
